src/preprocess.js

```diff
@@ -40,10 +122,104 @@ export const preprocess = text => {
 			return processor_options.ignore_styles(attrs) ? match.replace(/\S/g, ' ') : match;
 		});
 	}
-	// get information about the component
 	let result;
+	let processedResult;
+	let processedModule;
+	let processedInstance;
+	let processedStyle;
+	let processedMarkup;
+	let moduleExt = 'js';
+	let instanceExt = 'js';
 	try {
-		result = compiler.compile(text, { generate: false, ...processor_options.compiler_options });
+		// run preprocessor if present
+		if (processor_options.svelte_preprocess) {
+			const result = processor_options.svelte_preprocess(text, filename);
```

Comment:
If I interpret this correctly, this means that only synchronous preprocessing is supported? If so, this should be noted in the docs, too.

Reply:
@dummdidumm, thanks for your comment. Check this comment from an expert (lol):
https://github.com/sveltejs/eslint-plugin-svelte3/issues/10#issuecomment-490634346

Reply:
I know this comment, not sure what you mean. From the code it seems that the result has to return immediately, which is something people won't expect if they are used to async preprocessing, so it should be noted in the docs.

Reply:
Added a note in the README.md file